### ISSUE TITLE
Godot Editor NavigationRegion3D Bake Navmesh and NavigationRegion3D.bake function thru gdscript have different results

### ISSUE THREAD

**Issue by nightrobin** [godotengine/godot#39334]
**Godot version:**
4.0.dev
Commit: f05b4fe

**OS/device including version:**
Windows 10 x64 Bit

**Issue description:**
I created a scene with a floor and a box on top.
I use NavigationRegion3D.bake function to generate Navmesh, the whole floor is walkable, it does not take into account the box on top(the character can pass through the box, which should not).

I then tried to use the Godot Editor's Bake Navmesh, the box is taken into account and the character will not pass through the box, instead he will go around it.

**Expected Result:**
The character will not pass through the box, instead he will go around it.

**Steps to reproduce:**
Create a scene with a gdscript. Use NavigationRegion3D.bake function to generate Navmesh.

**Minimal reproduction project:**
[NavmeshIssue.zip](https://github.com/godotengine/godot/files/4739572/NavmeshIssue.zip)

**Comment by nightrobin** [comment]
Here a sample Navmesh outputs I generated

**Via Godot Editor NavigationRegion3D Bake Navmesh**

[gd_resource type="NavigationMesh" format=2]

[resource]
vertices = PackedVector3Array( 0.1, 0.9, 4.3, 0.1, 0.9, 1.3, -1.1, 0.9, 1, -4.4, 0.9, 4.3, 1, 0.9, -1.1, 1.3, 0.9, 0.1, 4.3, 0.9, 0.1, 4.3, 0.9, -4.4, -4.4, 0.9, 4.3, -1.1, 0.9, 1, -1.1, 0.9, -0.8, -4.4, 0.9, -4.4, 1, 0.9, -1.1, 4.3, 0.9, -4.4, -4.4, 0.9, -4.4, -1.1, 0.9, -0.8, 1, 0.9, 1.3, 0.1, 0.9, 1.3, 0.1, 0.9, 4.3, 4.3, 0.9, 0.1, 1.3, 0.9, 0.1, 1, 0.9, 1.3, 4.3, 0.9, 4.3, 4.3, 0.9, 0.1, 1, 0.9, 1.3, 0.1, 0.9, 4.3 )
polygons = [ PackedInt32Array( 1, 0, 2 ), PackedInt32Array( 2, 0, 3 ), PackedInt32Array( 5, 4, 6 ), PackedInt32Array( 6, 4, 7 ), PackedInt32Array( 9, 8, 10 ), PackedInt32Array( 10, 8, 11 ), PackedInt32Array( 12, 15, 13 ), PackedInt32Array( 13, 15, 14 ), PackedInt32Array( 18, 17, 16 ), PackedInt32Array( 21, 20, 19 ), PackedInt32Array( 25, 24, 22 ), PackedInt32Array( 22, 24, 23 ) ]



**Via GDScript NavigationRegion3D.bake function**

[gd_resource type="NavigationMesh" format=2]

[resource]
vertices = PackedVector3Array( -5, 0.5, 5, 5, 0.5, -5, 5, 0.5, 5, -5, 0.5, -5, -5, -0.5, 5, 5, -0.5, -5, 5, -0.5, 5, -5, -0.5, -5, 5, 0.5, 5, -5, 0.5, -5, 5, 0.5, -5, -5, 0.5, 5, 5, -0.5, 5, -5, -0.5, -5, 5, -0.5, -5, -5, -0.5, 5, 5, 0.5, 5, -5, -0.5, 5, -5, 0.5, 5, 5, -0.5, 5, 5, 0.5, -5, -5, -0.5, -5, -5, 0.5, -5, 5, -0.5, -5 )
polygons = [ PackedInt32Array( 0, 2, 4 ), PackedInt32Array( 2, 6, 4 ), PackedInt32Array( 1, 3, 5 ), PackedInt32Array( 3, 7, 5 ), PackedInt32Array( 8, 10, 12 ), PackedInt32Array( 10, 14, 12 ), PackedInt32Array( 9, 11, 13 ), PackedInt32Array( 11, 15, 13 ), PackedInt32Array( 16, 18, 20 ), PackedInt32Array( 18, 22, 20 ), PackedInt32Array( 17, 19, 21 ), PackedInt32Array( 19, 23, 21 ) ]

**Comment by Calinou** [comment]
cc @AndreaCatania

**Comment by smix8** [comment]
@nightrobin
I can't recreate the described issue in current Godot 4.x. Maybe your issue was fixed meanwhile or was result of a scripting bug.

I tried your test project but it is broken for current Godot 4.x. (Navigation node was replaced with NavigationServer, Cubemesh was removed, GDScript syntax changed, ...)

I created a new and very simple test project with a `NavigationRegion3D` and multiple `MeshInstances` as source geometry for baking.

I tried both editor `Bake NavMesh` button and `NavigationRegion.bake_navigation_mesh()`, and resulting .tres navigationmesh resources are identical in content.

@akien-mga 
I think issue can be closed.
